airtable/__init__.py

```diff
@@ -94,18 +95,29 @@ def __init__(self, base_id, api_key, dict_class=OrderedDict):
     def __request(self, method, url, params=None, payload=None):
         if method in ['POST', 'PUT', 'PATCH']:
             self.headers.update({'Content-type': 'application/json'})
+        
+        encoded_url = posixpath.join(self.base_url, urllib.parse.quote(url, safe=""))
```

Comment:
Please don't quote the whole url once you're here. Instead quote the table_name in the other functions